src/Solr/SolrIndex.php

```diff
@@ -779,6 +779,12 @@ public function search(SearchQuery $query, $offset = -1, $limit = -1, $params =
             foreach ($res->response->docs as $doc) {
                 $result = DataObject::get_by_id($doc->ClassName, $doc->ID);
                 if ($result) {
+                    // Filter out any results previously added to the solr index that have ShowInSearch == 0
+                    // This may also include any results where getShowInSearch() == false
+                    if (isset($result->ShowInSearch) && !$result->ShowInSearch) {
```

Comment:
This should note that it's a second level check, and we already prevent those records from being indexed in most cases. It also means the `$offset` and `$limit` args of this method might no longer be applicable, in the worst case you've asked for 20 results and you get zero because they've all been filtered out.

I think this is too low level, and should happen on the same level where we already do the canView() filtering (and devs already need to expect the issue raised above) - in `SearchForm->getResults()` and `CwpSearchEngine`. If you subclass that and replace the implementation you're on your own.

Reply:
>> It also means the $offset and $limit args of this method might no longer be applicable
I don't think there's anyway around this because we need to support getShowInSearch(), so we cannot rely solely on solr filtering which would allow us to preserve $offet and $limt.  This is really no different from how canView() has been all this time.

>> I think this is too low level
How come?  It seems like the only downside here is that we now make it really difficult for people to NOT run the ShowInSearch check

Reply:
We've talked through this on a call, results in https://github.com/silverstripe/silverstripe-fulltextsearch/issues/257#issuecomment-605717864

Reply:
I agree with @chillu about the placement of this code - better to have a consistent set of conditional logic incorporating both ShowInSearch and canView throughout the module. If you're using high-level APIs, you're covered, if you're using low-level APIs, the documentation should outline the risks and you have a clearer example to copy if the checks are neighboured.

Reply:
I think am in agreement here.  Reasoning is slightly different though.  I know there are plenty of custom search implementations in the wild that copy-pasted the SolrIndex->search() function into their own MyCustomSolrIndex because they want to tweak something e.g. highlighting.  Because of this, they don't call parent::search().  This means that any change we make here won't make it into their implementation.

Reply:
> they don't call parent::search(). This means that any change we make here won't make it into their implementation.

That's a risk I think we have to take, in alignment with the priorities I've outlined. We should avoid APIs that force devs to copy paste, and rely on composability instead (e.g. not forcing devs to mix in presentational highlighting concerns with core permission checks). Let's capture this in well written upgrade guides, outlining what happen before, what we've changed, and in which situations you could be vulnerable even after an upgrade.